File: data/input/bubble_sort2.cpp
#include <iostream>
#include <vector>

void sortWithBubble(std::vector<int>& data) {
    int length = data.size();
    
    for (int i = 0; i < length; i++) {
        bool hasSwapped = false;
        
        for (int j = 0; j < length - i - 1; j++) {
            if (data[j] > data[j + 1]) {
                std::swap(data[j], data[j + 1]);
                hasSwapped = true;
            }
        }
        
        if (!hasSwapped) {
            break;
        }
    }
}

int main() {
    std::vector<int> values = {64, 34, 25, 12, 22, 11, 90};
    
    std::cout << "Before sorting: ";
    for (int value : values) {
        std::cout << value << " ";
    }
    std::cout << std::endl;
    
    sortWithBubble(values);
    
    std::cout << "After sorting: ";
    for (int value : values) {
        std::cout << value << " ";
    }
    std::cout << std::endl;
    
    return 0;
}

File: data/input/BubbleSort2.java
public class BubbleSort2 {
    // Variante avec des noms différents
    public static void sortArray(int[] array) {
        int length = array.length;
        
        for (int i = 0; i < length; i++) {
            boolean hasSwapped = false;
            
            for (int j = 0; j < length - i - 1; j++) {
                if (array[j] > array[j + 1]) {
                    int temporary = array[j];
                    array[j] = array[j + 1];
                    array[j + 1] = temporary;
                    hasSwapped = true;
                }
            }
            
            if (!hasSwapped) {
                break;
            }
        }
    }
    
    public static void main(String[] args) {
        int[] data = {64, 34, 25, 12, 22, 11, 90};
        
        System.out.print("Before sorting: ");
        for (int value : data) {
            System.out.print(value + " ");
        }
        System.out.println();
        
        sortArray(data);
        
        System.out.print("After sorting: ");
        for (int value : data) {
            System.out.print(value + " ");
        }
        System.out.println();
    }
}
